src/main/java/seedu/duke/Duke.java

```diff
@@ -0,0 +1,784 @@
+     *
+     * @param index The index of the task to be removed.
+     * @return The removed task.
+     */
+    public Task removeTask(int index) {
+
+        return tasks.remove(index);
+    }
+    /**
+     * Retrieves a task from the task list at the specified index.
+     *
+     * @param index The index of the task to be retrieved.
+     * @return The task at the specified index.
+     */
+    public Task getTask(int index) {
+
+        return tasks.get(index);
+    }
+    /**
+     * Returns the number of tasks in the task list.
+     *
+     * @return The size of the task list.
+     */
+    public int getSize() {
+
+        return tasks.size();
+    }
+
+
+}
+
+
+enum TaskType {
+    TODO,
+    DEADLINE,
+    EVENT
+}
+
+class Task {
+    protected String description;
+    protected boolean isDone;
+    protected TaskType taskType;
+
+    public Task(String description, TaskType taskType) {
+        this.description = description;
+        this.isDone = false;
+        this.taskType = taskType;
+    }
+
+    public String getStatusIcon() {
+
+        return "[" + taskType + "]" + (isDone ? "[X] " : "[ ] ");
+    }
+
+    public String getDescription() {
+
+        return description;
+    }
+
+    public void markAsDone() {
+
+        isDone = true;
+    }
+
+    public void markAsNotDone() {
+
+        isDone = false;
+    }
+    @Override
+    public String toString() {
+
+        return getStatusIcon() + getDescription();
+    }
+}
+
+class Todo extends Task {
+    public Todo(String description) {
+
+        super(description, TaskType.TODO);
+    }
+}
+
+class Deadline extends Task {
+    protected LocalDateTime by;
+
+    public Deadline(String description, LocalDateTime by) {
+        super(description, TaskType.DEADLINE);
+        this.by = by;
+    }
+
+    @Override
+    public String getDescription() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy, HH:mm");
+        return super.getDescription() + " (by: " + formatter.format(by) + ")";
+    }
+    public LocalDateTime getBy() {
+
+        return by;
+    }
+}
+
+class Event extends Task {
+    protected LocalDateTime from;
+    protected LocalDateTime to;
+
+    public Event(String description, LocalDateTime from, LocalDateTime to) {
+        super(description, TaskType.EVENT);
+        this.from = from;
+        this.to = to;
+    }
+
+    @Override
+    public String getDescription() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy, HH:mm");
+        return super.getDescription() + " (from: " + formatter.format(from) + " to: " + formatter.format(to) + ")";
+    }
+    public LocalDateTime getFrom() {
+        return from;
+    }
+    public LocalDateTime getTo() {
+        return to;
+    }
+}
+/**
+ * Represents the main class for the Duke application.
+ * Initializes the application and starts the interaction with the user.
+ */
+public class Duke {
```

Comment:
Since the bot is no longer Duke should we rename the class to reflect the name of the chatbot?